projects/aca-content/src/lib/components/common/toggle-shared/toggle-shared.component.ts

```diff
@@ -22,48 +22,61 @@
  * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { Component, OnInit, Input } from '@angular/core';
-import { Observable } from 'rxjs';
+import { Component, OnInit, Input, OnDestroy } from '@angular/core';
+import { Observable, Subject } from 'rxjs';
 import { Store } from '@ngrx/store';
 import { SelectionState } from '@alfresco/adf-extensions';
 import { AppStore, ShareNodeAction, getAppSelection } from '@alfresco/aca-shared/store';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-toggle-shared',
   templateUrl: './toggle-shared.component.html'
 })
-export class ToggleSharedComponent implements OnInit {
+export class ToggleSharedComponent implements OnInit, OnDestroy {
   @Input()
   data: {
     iconButton?: string;
   };
 
   selection$: Observable<SelectionState>;
+  selectionState: SelectionState;
+  selectionLabel = '';
+  isShared = false;
+
+  onDestroy$ = new Subject<boolean>();
 
   constructor(private store: Store<AppStore>) {}
 
   ngOnInit() {
     this.selection$ = this.store.select(getAppSelection);
-  }
+    this.selection$.pipe(takeUntil(this.onDestroy$)).subscribe((selectionState) => {
+      this.selectionState = selectionState;
 
-  isShared(selection: SelectionState) {
-    // workaround for shared files
-    if (selection.first && selection.first.entry && (selection.first.entry as any).sharedByUser) {
-      return true;
-    }
+      // workaround for shared files
+      if (this.selectionState.first && this.selectionState.first.entry && (this.selectionState.first.entry as any).sharedByUser) {
+        this.isShared = true;
+      } else {
+        this.isShared =
```

Comment:
That whole if can be simplified to just: 
this.isShared = this.selectionState.first && this.selectionState.first.entry && (this.selectionState.first.entry as any).sharedByUser ? true : !!this.selectionState.first?.entry?.properties?.['qshare:sharedId']

Reply:
or even shorter: 
```ts
this.isShared =
        (this.selectionState.first && this.selectionState.first.entry && (this.selectionState.first.entry as any).sharedByUser) ||
        !!this.selectionState?.first?.entry?.properties?.['qshare:sharedId'];

Reply:
Replaced with 
```this.isShared =
        (this.selectionState.first && this.selectionState.first.entry && (this.selectionState.first.entry as any).sharedByUser) ||
        !!this.selectionState?.first?.entry?.properties?.['qshare:sharedId'];```